Wikipedia/Code/WMFExploreViewController.m

```diff
@@ -438,7 +484,7 @@ - (nullable UIView*)tableView:(UITableView*)tableView viewForFooterInSection:(NS
 - (void)tableView:(UITableView*)tableView willDisplayCell:(UITableViewCell*)cell forRowAtIndexPath:(NSIndexPath*)indexPath {
     id<WMFExploreSectionController> controller = [self sectionControllerForSectionAtIndex:indexPath.section];
 
-    if ([controller respondsToSelector:@selector(willDisplaySection)] && (![self isDisplayingCellsForSectionController:controller] || [self rowAtIndexPathIsOnlyRowVisibleInSection:indexPath])) {
```

Comment:
> That would seem to suggestion that the call to sendWillDisplayToVisibleSectionControllers when the app returns to the foreground isn't working.

I'll go back and double check this, but IIRC the specific case was first appearance of nearby section:

- Start the app, `viewDidAppear` on explore, only visible sections sent `willDisplay` (but nearby isn't visible yet)
- Scroll down to nearby
- Placeholder cells come on screen (willDisplay)
- Nearby results are fetched, table is reloaded (didEndDisplaying)
- "real" nearby cells are displayed (willDisplay), **but more than one is already on screen**
- nearby is "broken" (compass at 0deg, distance -3), since `willDisplaySection` is never called

This isn't a problem for other sections, because they're always sent `fetchIfNeeded`.  Now we're doing the same thing for "will display," which fixes the above issue (since it doesn't matter that more than 1 are displaying, we call `willDisplaySection` anyway)